include/netinet/in.h

```diff
@@ -217,26 +217,30 @@
 /* This macro to convert a 16/32-bit constant values quantity from host byte
  * order to network byte order.  The 16-bit version of this macro is required
  * for uIP:
- *
- *   Author Adam Dunkels <adam@dunkels.com>
- *   Copyright (c) 2001-2003, Adam Dunkels.
- *   All rights reserved.
  */
 
 #ifdef CONFIG_ENDIAN_BIG
 # define HTONS(ns) (ns)
 # define HTONL(nl) (nl)
 #else
+# ifdef __has_builtin && __has_builtin(__builtin_bswap16)
+#  define HTONS(ns) ((unsigned short)__builtin_bswap16(((unsigned short) (ns))))
```

Comment:
it's better to update the related macros in endian.h and forward HTONS to them.

Reply:
This is purely compiler dependent, so better to move compiler stuff to `compiler..h`

Reply:
Maybe adding something similar `CONFIG_HAVE_BUILTIN_CTZ` may be better.

Reply:
Done, moved to other headers